staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -431,6 +431,7 @@ func (s *Scheme) Convert(in, out interface{}, context interface{}) error {
 				return err
 			}
 			unstructuredOut.SetUnstructuredContent(content)
+			unstructuredOut.GetObjectKind().SetGroupVersionKind(gvk)
```

Comment:
Is there a reason why it isn't necessary to also set the GroupVersionKind here? https://github.com/atlassian/kubernetes/blob/cd8fd313bb4323d58e2d0965aa161267e2c2cd46/staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go#L452-L453

Reply:
There is a test that verifies it all works as expected (internal to unstructured) and it does (I just verified). So it is not necessary. The GVK fields that are picked up by the unstructured converter are set here
https://github.com/atlassian/kubernetes/blob/cd8fd313bb4323d58e2d0965aa161267e2c2cd46/staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go#L576

However I agree that it probably should not do that - if my reading of the code is correct, it sets the GVK for typed objects too, which it probably should not do.
@deads2k maybe you could clarify? Also there is some code above that calls `copyAndSetTargetKind()` - why does it need to set the GVK?